src/R/Editor/Impl/Formatting/FormatOperations.cs

```diff
@@ -23,8 +23,8 @@ internal static class FormatOperations {
         /// <summary>
         /// Formats statement that the caret is at
         /// </summary>
-        public static void FormatCurrentStatement(ITextView textView, ITextBuffer textBuffer) {
-            SnapshotPoint? caretPoint = MapCaretToBuffer(textView, textBuffer);
+        public static void FormatCurrentStatement(ITextView textView, ITextBuffer textBuffer, int caretOffset = 0) {
```

Comment:
Yes. This was because we were always formatting AFTER text buffer changed so after ENTER caret was on the next line and that led to -1 offset. Now we format BEFORE the change so default is 0. The only case we format after the change is on } (because of the way brace completion and provisional text work in the core editor)